app/models/pipeline_run.rb

```diff
@@ -1109,13 +1109,38 @@ def update_job_status
         # Check for long-running pipeline run and log/alert if needed
         check_and_log_long_run
       end
-      self.job_status = "#{prs.step_number}.#{prs.name}-#{prs.job_status}"
-      self.job_status += "|#{STATUS_READY}" if report_ready?
+      self.job_status = format_job_status_text(prs.step_number, prs.name, prs.job_status, report_ready?)
     end
     save!
     enqueue_new_pipeline_run if automatic_restart
```

Comment:
What happens to the automatic restart?

Reply:
Removed. This logic should be part of the step function.